src/receive_imf/receive_imf_tests.rs

```diff
@@ -2709,28 +2709,61 @@ async fn test_gmx_forwarded_msg() -> Result<()> {
     Ok(())
 }
 
-/// Tests that user is notified about new incoming contact requests.
+/// Tests that user is notified about new incoming contact requests,
+/// but not about additional messages arriving in the contact request chat.
 #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
 async fn test_incoming_contact_request() -> Result<()> {
-    let t = TestContext::new_alice().await;
+    let mut tcm = TestContextManager::new();
+    let alice = &tcm.alice().await;
+    let bob = &tcm.bob().await;
 
-    receive_imf(&t, MSGRMSG, false).await?;
-    let msg = t.get_last_msg().await;
-    let chat = chat::Chat::load_from_db(&t, msg.chat_id).await?;
+    let msg = tcm.send_recv(alice, bob, "Hello!").await;
+    let chat = chat::Chat::load_from_db(bob, msg.chat_id).await?;
     assert!(chat.is_contact_request());
 
-    let event = t
+    let event = bob
         .evtracker
         .get_matching(|evt| matches!(evt, EventType::IncomingMsg { .. }))
         .await;
     match event {
         EventType::IncomingMsg { chat_id, msg_id } => {
             assert_eq!(msg.chat_id, chat_id);
             assert_eq!(msg.id, msg_id);
-            Ok(())
         }
         _ => unreachable!(),
     }
+
+    // Bob ignores contact request.
+    // The second message does not result in notification.
+    let msg2 = tcm.send_recv(alice, bob, "Hello!!??").await;
+    let event = bob
+        .evtracker
+        .get_matching(|evt| matches!(evt, EventType::MsgsChanged { .. }))
+        .await;
+    match event {
+        EventType::MsgsChanged { chat_id, msg_id } => {
+            assert_eq!(msg2.chat_id, chat_id);
+            assert_eq!(msg2.id, msg_id);
+        }
+        _ => unreachable!(),
+    }
+
+    // Third message also does not result in the notification
```

Comment:
Do we need to test the event for the third message? If we need, this can be done in a loop